src/toil/jobStores/aws/jobStore.py

```diff
@@ -307,7 +307,8 @@ def _extractKeyInfoFromUrl(url):
         """
         s3 = boto.connect_s3()
         bucket = s3.get_bucket(url.netloc)
-        key = bucket.new_key(url.path[1:])
+        key = bucket.get_key(url.path[1:])
+        key = key if key is not None else bucket.new_key(url.path[1:])
```

Comment:
_extractKeyInfoFromUrl is also not an implementation of an abstract method since every jobstore has a different method for extracting data from URLs, so changing the signature does not have many repercussions